### PR TITLE
Implemented Canvas Downloading

### LINKED ISSUE #25: Canvas Downloading
Check if the proposed implementation of downloading the canvas as an image in #24 works on all browsers.

### PR DESCRIPTION
**Pull request by Elgayar**
Closes #25

### CODE REVIEW

**Comment by Elgayar** [comment]
Works for me on Chrome (87.0.4280.88), Edge (87.0.664.66), and Firefox (80.0.1).
Does not work on IE11

File: src/components/Board.vue [hunk 1]
```diff
@@ -127,6 +127,15 @@ export default Vue.extend({
 			const { width, height } = this.ctx.canvas;
 			this.ctx.clearRect(0, 0, width, height);
 		},
+
+		saveBoard() {
+			const url = this.ctx.canvas.toDataURL('image/png').replace(/^data:image\/[^;]/, 'data:application/octet-stream');
+		    let downloadLink = document.createElement('a');
+    		downloadLink.setAttribute('download', 'canvas.png');
```
Comment (aabounegm) [inline src/components/Board.vue:134]: Mix of tabs and spaces

File: src/components/Board.vue [hunk 2]
```diff
@@ -127,6 +127,15 @@ export default Vue.extend({
 			const { width, height } = this.ctx.canvas;
 			this.ctx.clearRect(0, 0, width, height);
 		},
+
+		saveBoard() {
+			const url = this.ctx.canvas.toDataURL('image/png').replace(/^data:image\/[^;]/, 'data:application/octet-stream');
+		    let downloadLink = document.createElement('a');
```
Comment (aabounegm) [inline src/components/Board.vue:133]: You can still use `const` here as the variable is not being reassigned

File: src/components/Board.vue [hunk 3]
```diff
@@ -127,6 +127,15 @@ export default Vue.extend({
 			const { width, height } = this.ctx.canvas;
 			this.ctx.clearRect(0, 0, width, height);
 		},
+
+		saveBoard() {
+			const url = this.ctx.canvas.toDataURL('image/png').replace(/^data:image\/[^;]/, 'data:application/octet-stream');
```
Comment (aabounegm) [inline src/components/Board.vue:132]:
```suggestion
		saveBoard(mimeType = 'image/png') {
			const url = this.ctx.canvas.toDataURL(mimeType).replace(/^data:image\/[^;]/, 'data:application/octet-stream');
```
To allow changing to `jpeg` or something

File: src/components/Board.vue [hunk 4]
```diff
@@ -127,6 +127,15 @@ export default Vue.extend({
 			const { width, height } = this.ctx.canvas;
 			this.ctx.clearRect(0, 0, width, height);
 		},
+
+		saveBoard() {
```
Comment (aabounegm) [inline src/components/Board.vue:131]: I think `downloadImage` is a more appropriate name

**Review by aabounegm — CHANGES_REQUESTED** [review]
I don't like the extra lines between the methods 😅
But great work overall

**Review by aabounegm — APPROVED** [review]

**Pull request merged**